woocommerce/woocommerce-ios
# Product Details: Basic information

Closes #856.

This PR displays basic information in the first section of the Product Details screen. Depending on if the product type is Grouped, Affiliate, Variable, or Basic, it will display an image, a title, total orders, review stats, and external link(s).

Since affiliate products aren't purchased directly through a user's store, I can't currently show a screenshot of this type of product in the app. It can be tested by overriding the code.
Hi-res still image:
<img src="https://user-images.githubusercontent.com/1062444/57881088-00232f00-77e6-11e9-9954-fe40b1518bd6.png" width="350" />

Animated gif:
![product-details-basics-section](https://user-images.githubusercontent.com/1062444/57880725-17155180-77e5-11e9-8b5e-49d5ff5faa69.gif)



Example of what the Status Badge will look like. (Ignore that it says "Publish" and imagine it says "Private" or "Draft".)
<img src="https://user-images.githubusercontent.com/1062444/57880452-6909a780-77e4-11e9-9a70-e9b274482807.png" width="350" />

## To test
- navigate to an order 
- fulfill the order
- select the product row
- wait for the product details view controller to load
- observe new rows (varies based on product type)

@kyleaparker some concerns: 
1. Is jumping out to Safari okay for the product permalink? I can change this to be a in-app web view (like Settings > Privacy Settings > Learn more does). I made it jump to safari in case they want to start editing the product in /wp-admin/. A web view has limitations (like poor JavaScript execution) that would slow the user down if the do access /wp-admin/ from that web page.

Update release notes:
